modules/gapi/include/opencv2/gapi/gmat.hpp

```diff
@@ -138,12 +187,25 @@ GAPI_EXPORTS GMatDesc descr_of(const cv::UMat &mat);
 /** @} */
 
 namespace gapi { namespace own {
-    class Mat;
     GAPI_EXPORTS GMatDesc descr_of(const Mat &mat);
 }}//gapi::own
 
 GAPI_EXPORTS std::ostream& operator<<(std::ostream& os, const cv::GMatDesc &desc);
 
+bool GMatDesc::canDescribe(const cv::gapi::own::Mat& mat) const
+{
+    const auto desc = planar ? descr_of(mat).asPlanar(chan) : descr_of(mat);
+    return *this == desc;
+}
+
+#if !defined(GAPI_STANDALONE)
+bool GMatDesc::canDescribe(const cv::Mat& mat) const
+{
+    const auto desc = planar ? descr_of(mat).asPlanar(chan) : descr_of(mat);
+    return *this == desc;
+}
+#endif
```

Comment:
Suggest to implement as a template with two instantiations (avoiding code duplication)

Reply:
Done